src/en/mangamo/src/eu/kanade/tachiyomi/extension/en/mangamo/cachedBy.kt

```diff
@@ -0,0 +1,22 @@
+package eu.kanade.tachiyomi.extension.en.mangamo
+
+import kotlin.reflect.KProperty
+
+@Suppress("ClassName")
+class cachedBy<T>(private val dependencies: () -> Any?, private val callback: () -> T) {
```

Comment:
why is this necessary?

Reply:
It allows properties to lazily update in response to changes in other properties. For example, it allows `auth` in `Mangamo.kt` to automatically re-create a `MangamoAuth` object and regenerate tokens when the user changes their user token in the settings.